### PR TITLE
Many important changes for 2.4

### PR DESCRIPTION
**Pull request by bsommerfeld**
Adding heuristic weights + fine tuned heuristic,
improved performance for over 50%, 
some small bugfixes,
Path is now an Iterable, improved javadocs, 
now using workstealing threads for async tasks + added shutdown process,
fixed getBlockState for lower version and implemented support for 1.18.2
added two new strategies for survival scenarios

### CODE REVIEW

File: pathetic-model/src/main/java/org/patheloper/model/pathing/pathfinder/AStarPathfinder.java [hunk 1]
```diff
@@ -357,4 +426,46 @@ private List<PathPosition> tracePathFromNode(Node endNode) {
     Collections.reverse(path); // make it the right order
     return path;
   }
+
+  /** Simple class to represent a 3D tuple as the HashMap key */
+  private class Tuple3<T> {
```
Comment (olijeffers0n) [inline pathetic-model/src/main/java/org/patheloper/model/pathing/pathfinder/AStarPathfinder.java]: Put this in some util package tbh. Doesn’t belong here

File: pathetic-model/src/main/java/org/patheloper/util/ExpiringHashMap.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
+package org.patheloper.util;
+
+import java.util.HashMap;
+import lombok.Getter;
+
+public class ExpiringHashMap<K, V> extends HashMap<K, ExpiringHashMap.Entry<V>> {
```
Comment (olijeffers0n) [inline pathetic-model/src/main/java/org/patheloper/util/ExpiringHashMap.java:6]: Pretty sure this already exists

File: pathetic-nms/v1_16/src/main/java/org/patheloper/nms/v1_16/OneSixteenNMSInterface.java [hunk 3]
```diff
@@ -25,4 +26,9 @@ public ChunkSnapshot getSnapshot(World world, int chunkX, int chunkZ) {
       return null;
     }
   }
+
+  @Override
+  public BlockState getBlockState(ChunkSnapshot snapshot, int x, int y, int z) {
+    return null;
```
Comment (olijeffers0n) [inline pathetic-nms/v1_16/src/main/java/org/patheloper/nms/v1_16/OneSixteenNMSInterface.java:32]: These need implementation

**Review by olijeffers0n — CHANGES_REQUESTED** [review]

File: pathetic-model/src/main/java/org/patheloper/util/ExpiringHashMap.java (hunk 2 above)
Comment (bsommerfeld) [reply pathetic-model/src/main/java/org/patheloper/util/ExpiringHashMap.java:6]: most likely, but its not worth getting an extra dependency for that

**Review by olijeffers0n — APPROVED** [review]

**Comment by olijeffers0n** [comment]
Untested, but looks good

**Pull request merged**